internal/builders/go/pkg/provenance.go

```diff
@@ -152,14 +150,20 @@ func GenerateProvenance(name, digest, command, envs, workingDir string, s signin
 	//
 	// NOTE: map is a reference so modifying invEnv modifies
 	// p.Predicate.Invocation.Environment.
-	invEnv := p.Predicate.Invocation.Environment.(map[string]interface{})
+	invEnv, ok := p.Predicate.Invocation.Environment.(map[string]interface{})
+	if !ok {
+		return nil, fmt.Errorf("converting %T to map[string]interface{}", p.Predicate.Invocation.Environment)
+	}
```

Comment:
I think it's ok to panic in this case. I guess the linter wants us to explicitly panic?

Reply:
Yes, this was reported by `errcheck` since we were not checking the `bool` returned from the type assertion. 

I see that it does not make sense to continue at this point if the assertion fails, but I'd prefer a nicely formatted error and the chance to do some cleanup over a panic. Would you like to `panic` instead of `fmt.Errorf`?

Reply:
In this case the error should not happen ever, so I think a panic is warranted.